oak/server/rust/oak_runtime/src/node/grpc_server.rs

```diff
@@ -167,16 +201,22 @@ impl GrpcServerNode {
                 error
             })?;
 
+        METRICS.requests_total.inc();
+        let timer = METRICS.request_duration.start_timer();
         // Create a gRPC request from an HTTP body.
-        Self::decode_grpc_request(&http_request_path, &http_request_body)
+        let res = Self::decode_grpc_request(&http_request_path, &http_request_body)
             // Process a gRPC request and send it into the Runtime.
             .and_then(|request| self.process_request(request))
             // Read a gRPC response from the Runtime.
             .and_then(|response_reader| self.process_response(response_reader))
             // Send gRPC response back to the HTTP client.
             .map(|body| Self::http_response(http::StatusCode::OK, body))
             // Convert an error to an HTTP response with a corresponding error status.
-            .or_else(|error| Ok(Self::http_response(error.into(), vec![])))
+            .or_else(|error| Ok(Self::http_response(error.into(), vec![])));
+
+        timer.observe_duration();
```

Comment:
Is this actually measuring any elapsed time? AFAICT everything is async here, so it's just measuring the time needed to create the `Future`. Or am I missing some magic happening behind the scenes in `async fn`?